File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.cc
#include <iostream>
#include <vector>
using namespace std;

string s, bomb;
vector<char> stack;

int main() {
    cin >> s;
    cin >> bomb;

    for (int i = 0; i < s.length(); i++) {
        stack.push_back(s[i]);

        if (stack.size() >= bomb.length()) {
            bool included = true;
            for (int j = 0; j < bomb.length(); j++) {
                if (stack[stack.size() - bomb.length() + j] != bomb[j]) {
                    included = false;
                    break;
                }
            }

            if (included) {
                for (int j = 0; j < bomb.length(); j++) {
                    stack.pop_back();
                }
            }
        }
    }

    if (stack.empty()) {
        cout << "FRULA";
    } else {
        for (char c : stack) {
            cout << c;
        }
    }
    return 0;
}

File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.py
string = input()
bomb = input()

stack = []
bomb_len = len(bomb)

for i in range(len(string)):
    stack.append(string[i])
    if ''.join(stack[-bomb_len:]) == bomb:
        for _ in range(bomb_len):
            stack.pop()

if stack == []:
    print("FRULA")
else:
    print(''.join(stack))